### DISCUSSION TITLE
How does vue3 get the current page route instance in non-vue file？

### DISCUSSION THREAD

**Discussion by alamhubb** [vuejs/router#1872] [Help and Questions] [answered]
I want to develop a utility class. The tool class provides a method to return the current page routing instance. How to achieve?
useRouter doesn't work in non-Vue files

I saw a way to rely on store implementation. Is there any function that does not depend on third-party implementation? Why is useRouter limited to be used only on vue component pages?

**Comment by posva** [comment, marked as answer]
useRouter, like other composables, depend on `inject()` and thus can only be used within `setup()`. If you are not doing SSR, you can just import the `router` instance
